pkg/config/servicebuild.go

```diff
@@ -108,129 +109,259 @@ func (bldr *ServiceBuilder) WithDataService() *ServiceBuilder {
+		awsSession, err := bldr.Session()
+		if err != nil {
+			return err
+		}
+
+		bldr.WithService(&common.S3{
+			Client:  s3.New(awsSession),
+			Manager: s3manager.NewDownloader(awsSession),
+		})
+
+		return nil
+	})
+	return bldr
+}
+
+func (bldr *ServiceBuilder) WithNotificationer() *ServiceBuilder {
+	bldr.handlers = append(bldr.handlers, func() error {
+		awsSession, err := bldr.Session()
+		if err != nil {
+			return err
+		}
+
+		bldr.WithService(&common.SNS{
+			Client: sns.New(awsSession),
+		})
+
+		return nil
+	})
+	return bldr
+}
+
 // Build creates and returns a structue with AWS services
-func (bldr *ServiceBuilder) Build() (*ConfigurationBuilder, error) {
-	err := bldr.Config.Build()
+func (bldr *ServiceBuilder) Build() (*ServiceBuilder, error) {
+	err := bldr.ConfigurationBuilder.Build()
 	if err != nil {
 		// We failed to build the configuration, so honestly there is no
 		// point in continuating...
-		return bldr.Config, AWSConfigurationError(err)
+		return bldr, AWSConfigurationError(err)
 	}
 
 	// Create session is done first, and explicitly, because everything else
 	// uses it
-	err = bldr.createSession(bldr.Config)
+	err = bldr.createSession()
 
 	if err != nil {
 		log.Printf("Could not create session: %s", err.Error())
-		return bldr.Config, AWSConfigurationError(err)
+		return bldr, AWSConfigurationError(err)
 	}
 
 	for _, f := range bldr.handlers {
-		err := f(bldr.Config)
+		err := f()
 		if err != nil {
 			log.Printf("Error while trying to execute handler: %s", runtime.FuncForPC(reflect.ValueOf(f).Pointer()).Name())
-			return bldr.Config, AWSConfigurationError(err)
+			return bldr, AWSConfigurationError(err)
 		}
 	}
 
 	// Setting config values from parameter store requires services to be configured first
-	if err := bldr.Config.RetrieveParameterStoreVals(); err != nil {
-		return bldr.Config, AWSConfigurationError(err)
+	if err := bldr.RetrieveParameterStoreVals(); err != nil {
+		return bldr, AWSConfigurationError(err)
 	}
 
 	// make certain build is called before returning.
-	return bldr.Config, nil
+	return bldr, nil
 }
 
-func (bldr *ServiceBuilder) createSession(config ConfigurationServiceBuilder) error {
+func (bldr *ServiceBuilder) createSession() error {
 	var err error
-	region, err := bldr.Config.GetStringVal("AWS_CURRENT_REGION")
-	if err == nil {
+	region := common.GetEnv("AWS_CURRENT_REGION", "")
+	var awsSession *session.Session
+	if region == "" {
 		log.Printf("Using AWS region \"%s\" to create session...", region)
-		bldr.awsSession, err = session.NewSession(
+		awsSession, err = session.NewSession(
 			&aws.Config{
 				Region: aws.String(region),
 			},
 		)
+		if err != nil {
+			return err
+		}
 	} else {
 		log.Println("Creating AWS session using defaults...")
-		bldr.awsSession, err = session.NewSession()
+		awsSession, err = session.NewSession()
+		if err != nil {
+			return err
+		}
 	}
-	return err
+
+	bldr.WithService(awsSession)
```

Comment:
Define the AwsSession as a service, so consumers can access it directly
(which the reset codebuild needs)